src/user_programs/user_program_led_matrix.cpp

```diff
@@ -0,0 +1,81 @@
+#include "hal/gpio_wrapper.hpp"
+#include "microbit_v2.h"
+#include "scheduler/pending_process_callbacks.hpp"
+#include "userlib/syscalls.hpp"
+
+#include <stdint.h>
+
+#include <cstdio>
+
+static uint8_t row_index = 0;
```

Comment:
Can some of these be non-static (moved into stack on led_matrix_task) or at least static function variables?

Reply:
all on stack now